doc/nrf/links.txt

```diff
@@ -1883,7 +1883,7 @@
 .. _`nRF54H20 SoC binaries v0.9.2`: https://files.nordicsemi.com/ui/native/SDSC/external/nrf54h20_soc_binaries_v0.9.2.zip
 .. _`nRF54H20 SoC binaries v0.9.6`: https://files.nordicsemi.com/ui/native/SDSC/external/nrf54h20_soc_binaries_v0.9.6.zip
 
-.. _`nRF54H20 SoC binaries v22.2.0+14`: https://eu.files.nordicsemi.com/ui/native/SDSC/external-confidential/nrf54h20_soc_binaries_v22.2.0+14.zip
+.. _`nRF54H20 SoC binaries v22.2.0+14`: https://files.nordicsemi.com/artifactory/SDSC/external/nrf54h20_soc_binaries_v22.2.0%2B14.zip
```

Comment:
Could we mention this link here https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_ironside.rst?plain=1#L24 if it is for IronSide?